src/components/molecules/NavSearchBar/NavSearchBar.tsx

```diff
@@ -141,6 +143,20 @@ const NavSearchBar = () => {
     [searchQuery]
   );
 
+  const focusSearchBar = useCallback((e) => {
+    e.preventDefault();
+    const searchBar = document.getElementById("NavSearchBar__search-input");
+    if (searchBar) searchBar.focus();
+    // TODO: we need to remove "f" from being entered into the input which would happen now
+  }, []);
```

Comment:
Using `document.getElementById` isn't a very Reacty way of doing things. We would likely want to get a `ref` for that element if we were wanting to call `.focus()` on it directly:

- https://reactjs.org/docs/refs-and-the-dom.html
- https://reactjs.org/docs/hooks-reference.html#useref
- https://reactjs.org/docs/forwarding-refs.html

Reply:
> TODO: we need to remove "f" from being entered into the input which would happen now

This part is interesting, and something I was going to call out in my review to ask how you were handling that case. Not sure of the solution off the top of my head just now.

---

Actually, looking at the docs, it looks like it should handle this case by default:

- https://craig.is/killing/mice#api.bind.text-fields
  - > By default all keyboard events will not fire if you are inside of a `textarea`, `input`, or `select` to prevent undesirable things from happening.
  - > If you want them to fire you can add the class `mousetrap` to the element.
- https://craig.is/killing/mice#api.stopCallback
  - > `Mousetrap.stopCallback` is a method that is called to see if the keyboard event should fire based on if you are inside a form `input` field.

Is that not the behaviour you're observing here?

Reply:
I don't think so -- the issue wasn't  about being in the input field already to handle an event but to prevent input field from getting that initial `f`.  @soichih solved it in b75e4d6fb97b0695c57eee8a7107004807aba9f5 by 
```
--- a/src/components/molecules/NavSearchBar/NavSearchBar.tsx
+++ b/src/components/molecules/NavSearchBar/NavSearchBar.tsx
@@ -144,6 +144,7 @@ const NavSearchBar = () => {
   );
 
   const focusSearchBar = useCallback((e) => {
+    e.preventDefault();
```

Reply:
> Using `document.getElementById` isn't a very Reacty way of doing things. We would likely want to get a `ref` for that element if we were wanting to call `.focus()` on it directly:

ignorant in React me tried and failed... I might give it another shot later on

Reply:
<details>
<summary>here is what I have tried</summary> 

```patch
diff --git a/src/components/molecules/NavSearchBar/NavSearchBar.tsx b/src/components/molecules/NavSearchBar/NavSearchBar.tsx
index 674f3dc4..68ff5f25 100644
--- a/src/components/molecules/NavSearchBar/NavSearchBar.tsx
+++ b/src/components/molecules/NavSearchBar/NavSearchBar.tsx
@@ -58,6 +58,7 @@ const NavSearchBar = () => {
   const { worldUsers } = useWorldUsers();
 
   const { openUserProfileModal } = useProfileModalControls();
+  const searchBar = React.useRef(null);
 
   const foundRooms = useMemo<JSX.Element[]>(() => {
     if (!searchQuery) return [];
@@ -148,8 +149,7 @@ const NavSearchBar = () => {
 
   const focusSearchBar = useCallback((e) => {
     e.preventDefault();
-    const searchBar = document.getElementById("NavSearchBar__search-input");
-    if (searchBar) searchBar.focus();
+    this.searchBar.current.focus();
   }, []);
 
   useMousetrap({
@@ -179,7 +179,7 @@ const NavSearchBar = () => {
       <InputField
         value={searchInputValue}
         inputClassName="NavSearchBar__search-input"
-        id="NavSearchBar__search-input"
+        ref={searchBar}
         onChange={onSearchInputChange}
         placeholder="Search for people, rooms, events..."
         autoComplete="off"
```
</details>
I get
```
/home/yoh/proj/misc/sparkle/sparkle/src/components/molecules/NavSearchBar/NavSearchBar.tsx
TypeScript error in /home/yoh/proj/misc/sparkle/sparkle/src/components/molecules/NavSearchBar/NavSearchBar.tsx(152,5):
Object is possibly 'undefined'.  TS2532

    150 |   const focusSearchBar = useCallback((e) => {
    151 |     e.preventDefault();
  > 152 |     this.searchBar.current.focus();
        |     ^
    153 |   }, []);
    154 | 
    155 |   useMousetrap({
```
how could I improve?  @0xdevalias @soichih?

Reply:
Merged @soichih fix. Should be all set

Reply:
> how could I improve

@yarikoptic The issue in that code snippet you provided is that you're trying to use `this.`, which references object state within a class. Yet we use functional components.

Since your `const searchBar = React.useRef(null);` is within the same 'variable scope' as where you want to use it, you can just use `searchBar` directly (no `this.`)

The only other issue is that it could be `null`/`undefined`, so you would use TypeScript's 'safe accessor'/'optional chaining' syntax. Something like: `searchBar.current?.focus();`

You would also need to ensure you add `searchBar` to the 'deps' array of the `useCallback` so that it knows to break the memoisation when that variable changes.

---

> Merged @soichih fix. Should be all set

Sounds good :)